src/java.base/share/classes/java/lang/foreign/MemorySession.java

```diff
@@ -215,14 +215,17 @@ public sealed interface MemorySession extends AutoCloseable, SegmentAllocator pe
 
      /**
      * Creates a native memory segment with the given size (in bytes), alignment constraint (in bytes) associated with
-      * this memory session.
+     * this memory session. The {@link MemorySegment#address()} of the returned memory segment is the starting address of
+     * the newly allocated off-heap memory region backing the segment. Moreover, the {@linkplain MemorySegment#address() address}
+     * of the returned segment will be aligned according the provided alignment constraint.
+     * <p>
      * A client is responsible for ensuring that this memory session is closed when the
      * segment is no longer in use. Failure to do so will result in off-heap memory leaks.
      * <p>
-     * The off-heap memory associated with the returned native memory segment is initialized to zero.
+     * The off-heap region of memory associated with the returned native memory segment is initialized to zero.
      *
      * @param byteSize the size (in bytes) of the off-heap memory block backing the native memory segment.
-     * @param byteAlignment the alignment constraint (in bytes) of the off-heap memory block backing the native memory segment.
+     * @param byteAlignment the alignment constraint (in bytes) of the off-heap region of memory backing the native memory segment.
```

Comment:
well spotted - "backing" is the way to go. Memory sessions are "associated" to segments. Regions of memory "back" segments. Note that we deliberately use "region of memory" instead of "memory region" to avoid confusion between the concept and API construct (e.g. memory segment - there are many memory XYZ in the API).